langchain_workflow.py

```
@@ -1,46 +1,35 @@
+def divide(x,y):
+ if y==0: return "Error! Division by zero."
+ return x/y
 
 def calculator():
-    print("Welcome to the Simple Calculator!")
-    print("Select operation:")
-    print("1. Add")
-    print("2. Subtract")
-    print("3. Multiply")
-    print("4. Divide")
-    print("5. Exit")
-
-    while True:
-        choice = input("Enter choice (1/2/3/4/5): ")
-
-        if choice in ['1', '2', '3', '4']:
-            num1 = float(input("Enter first number: "))
-            num2 = float(input("Enter second number: "))
-
-            if choice == '1':
-                print(f"{num1} + {num2} = {add(num1, num2)}")
-            elif choice == '2':
-                print(f"{num1} - {num2} = {subtract(num1, num2)}")
-            elif choice == '3':
-                print(f"{num1} * {num2} = {multiply(num1, num2)}")
-            elif choice == '4':
-                print(f"{num1} / {num2} = {divide(num1, num2)}")
-        elif choice == '5':
-            print("Exiting the calculator. Goodbye!")
-            break
-        else:
-            print("Invalid input. Please enter a number between 1 and 5.")
-
-if __name__ == "__main__":
-    calculator()
+ print("Welcome to the Simple Calculator!")
+ print("Select operation:")
+ print("1. Add")
+ print("2. Subtract")
+ print("3. Multiply")
+ print("4. Divide")
+ print("5. Exit")
+ while True:
+  choice=input("Enter choice (1/2/3/4/5): ")
+  if choice in ['1','2','3','4']:
+   num1=float(input("Enter first number: "))
+   num2=float(input("Enter second number: "))
+   if choice=='1':
+    print(f"{num1} + {num2} = {add(num1,num2)}")
+   elif choice=='2':
+    print(f"{num1} - {num2} = {subtract(num1,num2)}")
+   elif choice=='3':
+    print(f"{num1} * {num2} = {multiply(num1,num2)}")
+   elif choice=='4':
+    print(f"{num1} / {num2} = {divide(num1,num2)}")
+  elif choice=='5':
+   print("Exiting the calculator. Goodbye!")
+   break
+  else:
+   print("Invalid input. Please enter a number between 1 and 5.")
+
```

This line exceeds the recommended maximum line length of 79 characters. Consider breaking it into multiple lines.  Also, it might be better to raise a ValueError exception instead of printing an error message. This allows for better error handling elsewhere in the program.